x-pack/plugins/global_search_providers/public/index.ts

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { PluginInitializer } from 'src/core/public';
+import {
+  GlobalSearchProvidersPlugin,
+  GlobalSearchProvidersPluginSetupDeps,
+  GlobalSearchProvidersPluginStartDeps,
+} from './plugin';
+import { GlobalSearchProvidersPluginSetup, GlobalSearchProvidersPluginStart } from './types';
+
+export const plugin: PluginInitializer<
+  GlobalSearchProvidersPluginSetup,
+  GlobalSearchProvidersPluginStart,
+  GlobalSearchProvidersPluginSetupDeps,
+  GlobalSearchProvidersPluginStartDeps
+> = (context) => new GlobalSearchProvidersPlugin();
```

Comment:
nit: you could just leave these generics empty. Honestly they don't really provide any value, maybe we should just remove them from the PluginInitializer type.

Reply:
My java background probably. Will remove them.